controllers/control/cassandratask_controller_test.go

```diff
@@ -158,254 +223,361 @@ func waitForTaskCompletion(taskKey types.NamespacedName) *api.CassandraTask {
-			Expect(callDetails.URLCounts["/api/v0/ops/executor/job"]).To(Equal(0))
-			Expect(callDetails.URLCounts["/api/v0/metadata/versions/features"]).To(BeNumerically(">", 1))
-
-			// verifyPodsHaveAnnotations(testNamespaceName, string(task.UID))
-			Expect(completedTask.Status.Succeeded).To(BeNumerically(">=", 1))
-		})
-
-		It("Runs a upgradesstables task against the datacenter pods", func() {
-			By("Creating a task for upgradesstables")
-			time.Sleep(1 * time.Second) // Otherwise the CreationTimestamp could be too new
-			taskKey := createTask(api.CommandUpgradeSSTables, testNamespaceName)
-
-			completedTask := waitForTaskCompletion(taskKey)
-
-			Expect(callDetails.URLCounts["/api/v0/ops/tables/sstables/upgrade"]).To(Equal(3))
-			Expect(callDetails.URLCounts["/api/v0/ops/executor/job"]).To(Equal(0))
-			Expect(callDetails.URLCounts["/api/v0/metadata/versions/features"]).To(BeNumerically(">", 1))
-
-			// verifyPodsHaveAnnotations(testNamespaceName, string(task.UID))
-			Expect(completedTask.Status.Succeeded).To(BeNumerically(">=", 1))
+			deleteDatacenter(testNamespaceName)
+			// Expect(k8sClient.Delete(context.TODO(), testDc)).Should(Succeed())
 		})
 
-		It("Replaces a node in the datacenter", func() {
-			By("Creating a task for replacenode")
-			time.Sleep(1 * time.Second) // Otherwise the CreationTimestamp could be too new
-			taskKey, task := buildTask(api.CommandReplaceNode, testNamespaceName)
-
-			podKey := types.NamespacedName{
-				Name:      fmt.Sprintf("test-cassdc-%s-pod%d", testDatacenterName, 2),
-				Namespace: testNamespaceName,
-			}
-
-			task.Spec.Jobs[0].Arguments = map[string]string{"pod_name": podKey.Name}
-			Expect(k8sClient.Create(context.TODO(), task)).Should(Succeed())
+		Context("Restart", func() {
+			It("Restarts a single rack", func() {
+				stsKey := types.NamespacedName{Namespace: testNamespaceName, Name: fmt.Sprintf("%s-%s-r1-sts", clusterName, testDc.Name)}
+				var sts appsv1.StatefulSet
+				Expect(k8sClient.Get(context.TODO(), stsKey, &sts)).Should(Succeed())
 
-			// Verify the pod2 was deleted
-			Eventually(func() bool {
-				pod := &corev1.Pod{}
-				err := k8sClient.Get(context.TODO(), podKey, pod)
-				return err != nil && errors.IsNotFound(err)
-			}, time.Duration(3*time.Second)).Should(BeTrue())
+				// Create task to restart r1
+				taskKey, task := buildTask(api.CommandRestart, testNamespaceName)
+				task.Spec.Jobs[0].Arguments = map[string]string{"rack": "r1"}
+				Expect(k8sClient.Create(context.Background(), task)).Should(Succeed())
 
-			// Recreate it so the process "finishes"
-			createPod(testNamespaceName, fmt.Sprintf("test-%s", testDatacenterName), testDatacenterName, 2)
+				Eventually(func() bool {
+					Expect(k8sClient.Get(context.TODO(), stsKey, &sts)).Should(Succeed())
+					_, found := sts.Spec.Template.ObjectMeta.Annotations[api.RestartedAtAnnotation]
+					return found
+				}).Should(BeTrue())
+
```

Comment:
You could maybe add a check here to see if the completion time is still `nil`.

Reply:
Done.